### PR TITLE
Makes deps that require torch optional by default

### PR DESCRIPTION
**Pull request by binaryaaron**
## What's changing
As discussed, this changes the dependency structure of this repo to make all packages that require Torch optional by default. This allows reuse of the configs more readily without other structural changes.

## How to test it
Tested internally by generating lockfiles on a project requiring this project with different  combos of [torchy, test, docs], verified pytorch is only in the ones in `torchy`.

### CODE REVIEW

**Comment by binaryaaron** [comment]
As a note, we can fix the flakey integration test - will do in a distinct PR.

File: pyproject.toml [hunk 1]
```diff
@@ -20,18 +20,25 @@ requires-python = ">=3.11,<3.12"
...
     "pydantic-yaml==1.2.0",
     "ray[default]==2.30.0",
     "loguru==0.7.2",
     "s3fs",
-    # HuggingFace
+]
+
+[project.optional-dependencies]
+ruff = ["ruff==0.2.1"]
+torchy = [
```
Comment (veekaybee) [inline pyproject.toml:39]: maybe `jobs`  since it's `evaluation` + `finetuning`?

**Review by veekaybee — APPROVED** [review]
👍 lgtm

File: pyproject.toml (hunk 1 above)
Comment (binaryaaron) [reply pyproject.toml:39]: i suppose that is indeed better than `torchy`. :)

**Pull request merged**